File: apps/ui/src/features/bot/types/provider/contract/update.ts
// apps/ui/src/features/bot/types/provider/contract/update.ts
// 内部引用
import type { ProviderId } from "../common";
import type { MutationResponse } from "./response";

/** 对应 Rust UpdateEnabledModelsRequest */
export interface UpdateEnabledModelsPayload {
  providerId: ProviderId;
  enabledModels: string[];
}

/** 对应 Rust UpdateEnabledModelsResponse */
export type UpdateEnabledModelsResponse = MutationResponse;

File: apps/desktop/src-tauri/src/core/bot/models/provider/contract/update.rs
// apps/desktop/src-tauri/src/core/bot/models/provider/contract/update.rs
use serde::Deserialize;

use super::super::ProviderId;
use super::ProviderCommandResponse;

/// Request payload for updating enabled models.
///
/// 更新已启用模型列表的请求载荷。
#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct UpdateEnabledModelsRequest {
    pub provider_id: ProviderId,
    pub enabled_models: Vec<String>,
}

/// Response for updating enabled models.
///
/// 更新已启用模型列表的响应。
pub(crate) type UpdateEnabledModelsResponse = ProviderCommandResponse;
